### ISSUE TITLE
Unable to discover proxmox disks

### ISSUE THREAD

**Issue by AmitKumarDas** [openebs/openebs#2704]
This issue captures the slack conversations w.r.t inability of NDM to discover certain disks.

:question: Hello all,
I continue my journey to understand openebs storage.
Reading the documentation regarding the cStor I added 8 * 32G disks to each nodes in my cluster. Actually there's 3 nodes in my cluster so I assume that I will seen 8 * 3 = 24 blockdevices in my clusters. But there's only 8:

```bash
$ kubectl get disk
NAME                                    SIZE          STATE    AGE
disk-10cbf5d6bd8330920fcb3bdda1aa6a55   34359738368   Active   35m
disk-11ebb418e80395b0378eca084e2a17e4   34359738368   Active   35m
disk-169068a2a396a22759d5bfa04f4a1b04   34359738368   Active   18m
disk-51e5c068c21f5186044957960dfbaefb   34359738368   Active   36m
disk-9d0e2efbf255eec778874a9fd42f1a72   34359738368   Active   31m
disk-c2b57cce00020b1d1562b9c469a949fa   34359738368   Active   31m
disk-dae3cec149cfd5fc5c6826caa0650e21   34359738368   Active   31m
disk-db1254ebd777a99e6b9b5626358c7038   34359738368   Active   17m
```

It seems the 8 disks are dispatched between two nodes k8snodv01p and k8snodv03p:
```bash
$ kubectl describe disk | grep -A3 "Name:         disk-"|grep "Labels:       kubernetes.io/hostname="|sort
Labels:       kubernetes.io/hostname=k8snodv01p
Labels:       kubernetes.io/hostname=k8snodv01p
Labels:       kubernetes.io/hostname=k8snodv01p
Labels:       kubernetes.io/hostname=k8snodv01p
Labels:       kubernetes.io/hostname=k8snodv03p
Labels:       kubernetes.io/hostname=k8snodv03p
Labels:       kubernetes.io/hostname=k8snodv03p
Labels:       kubernetes.io/hostname=k8snodv03p
```

:confused: Does that mean that no disks are seen from my second node k8snodv02p ?
Sorry to ask that question maybe it's very simple but how can I ensure all my 24 disks are seen from openebs perspective ? 

:question: Sai Chaithanya:mayadata_m: 
Is ndm daemon pod  running on nodev02p node? (edited)

:bulb: Wilf1rst:
The ndm pod running on each node
```bash
$ kubectl -n openebs get pods -o wide
NAME                                           READY   STATUS    RESTARTS   AGE   IP           NODE         NOMINATED NODE   READINESS GATES
openebs-admission-server-797586bf87-w9s54      1/1     Running   0          14h   10.42.2.56   k8snodv03p   <none>           <none>
openebs-apiserver-5c87d744b8-sff4f             1/1     Running   0          14h   10.42.2.57   k8snodv03p   <none>           <none>
openebs-localpv-provisioner-6f74f55454-rwpkm   1/1     Running   1          14h   10.42.3.39   k8snodv01p   <none>           <none>
openebs-ndm-2qkj6                              1/1     Running   0          14h   172.16.0.7   k8snodv01p   <none>           <none>
openebs-ndm-fg7z5                              1/1     Running   0          14h   172.16.0.8   k8snodv02p   <none>           <none>
openebs-ndm-fhflg                              1/1     Running   0          14h   172.16.0.9   k8snodv03p   <none>           <none>
openebs-ndm-operator-55cc69dc96-mdplv          1/1     Running   0          14h   10.42.2.58   k8snodv03p   <none>           <none>
openebs-provisioner-78b964ffb9-kk46k           1/1     Running   1          14h   10.42.3.40   k8snodv01p   <none>           <none>
openebs-snapshot-operator-56c69dcb58-sw4vj     2/2     Running   1          14h   10.42.1.47   k8snodv02p   <none>           <none>
```

:spiral_notepad: Also there's a view of my disk from my hosts perspective (I just check and it's the same on my 3 k8s hosts):
```bash
$ sudo fdisk -l|grep "^Disk"
Disk /dev/sda: 107.4 GB, 107374182400 bytes, 209715200 sectors
Disk label type: dos
Disk identifier: 0x000b956b
Disk /dev/sdb: 34.4 GB, 34359738368 bytes, 67108864 sectors
Disk /dev/sdc: 34.4 GB, 34359738368 bytes, 67108864 sectors
Disk /dev/sdd: 34.4 GB, 34359738368 bytes, 67108864 sectors
Disk /dev/sde: 34.4 GB, 34359738368 bytes, 67108864 sectors
Disk /dev/sdf: 34.4 GB, 34359738368 bytes, 67108864 sectors
Disk /dev/sdg: 34.4 GB, 34359738368 bytes, 67108864 sectors
Disk /dev/sdh: 34.4 GB, 34359738368 bytes, 67108864 sectors
Disk /dev/sdi: 34.4 GB, 34359738368 bytes, 67108864 sectors
```

Wilf1rst  18 hours ago
Here's the last line logs from the pods openebs-ndm-fg7z5 on k8snov02p:
```
:"QEMU_HARDDISK", Serial:"drive-scsi8", Vendor:"QEMU", Path:"/dev/sdi", ByIdDevLinks:[]string{"/dev/disk/by-id/scsi-0QEMU_QEMU_HARDDISK_drive-scsi8"}, ByPathDevLinks:[]string{"/dev/disk/by-path/pci-0000:00:05.0-scsi-0:0:0:8"}, FirmwareRevision:"", LogicalSectorSize:0x200, PhysicalSectorSize:0x0, Compliance:"", DeviceType:"", PartitionType:"", FileSystemInfo:controller.FSInfo{FileSystem:"None", MountPoint:""}}
I0810 10:44:20.679292       6 blockdevicestore.go:98] Updated blockdevice object : blockdevice-db1254ebd777a99e6b9b5626358c7038
```
It seems that a disk is detected.

:question: Sai Chaithanya:mayadata_m:
Can you pasted the output of kubectl get bd -n openebs

:bulb: Wilf1rst 
Here it is:
```bash
$ kubectl get bd -n openebs
NAME                                           SIZE          CLAIMSTATE   STATUS   AGE
blockdevice-10cbf5d6bd8330920fcb3bdda1aa6a55   34359738368   Unclaimed    Active   62m
blockdevice-11ebb418e80395b0378eca084e2a17e4   34359738368   Unclaimed    Active   62m
blockdevice-169068a2a396a22759d5bfa04f4a1b04   34359738368   Unclaimed    Active   45m
blockdevice-51e5c068c21f5186044957960dfbaefb   34359738368   Unclaimed    Active   63m
blockdevice-9d0e2efbf255eec778874a9fd42f1a72   34359738368   Unclaimed    Active   58m
blockdevice-c2b57cce00020b1d1562b9c469a949fa   34359738368   Unclaimed    Active   58m
blockdevice-dae3cec149cfd5fc5c6826caa0650e21   34359738368   Unclaimed    Active   58m
blockdevice-db1254ebd777a99e6b9b5626358c7038   34359738368   Unclaimed    Active   44m
sparse-74a30cf9fb7f5f8dc8be1f3c5cb5bed3        10737418240   Unclaimed    Active   14h
sparse-a29c08ed690c05073f0e149247a620d9        10737418240   Unclaimed    Active   14h
sparse-ddea1324282dfec8b75645302e8b54b7        10737418240   Unclaimed    Active   14h
```

:spiral_notepad: Regarding what I said previously that a disk is detected on my k8snodv02p ndm pod, it seems that the blockdevice is from my k8snodv01p:
```yaml
$ kubectl -n openebs describe bd blockdevice-db1254ebd777a99e6b9b5626358c7038
Name:         blockdevice-db1254ebd777a99e6b9b5626358c7038
Namespace:    openebs
Labels:       kubernetes.io/hostname=k8snodv01p
              ndm.io/blockdevice-type=blockdevice
              ndm.io/managed=true
Annotations:  <none>
API Version:  openebs.io/v1alpha1
Kind:         BlockDevice
Metadata:
  Creation Timestamp:  2019-08-10T10:43:45Z
  Generation:          1
  Resource Version:    2261226
  Self Link:           /apis/openebs.io/v1alpha1/namespaces/openebs/blockdevices/blockdevice-db1254ebd777a99e6b9b5626358c7038
  UID:                 ba9b82df-bb5b-11e9-8221-0200005283a7
Spec:
  Capacity:
    Logical Sector Size:   512
    Physical Sector Size:  0
    Storage:               34359738368
  Details:
    Compliance:         
    Device Type:        
    Firmware Revision:  
    Model:              QEMU_HARDDISK
    Serial:             drive-scsi8
    Vendor:             QEMU
  Devlinks:
    Kind:  by-id
    Links:
      /dev/disk/by-id/scsi-0QEMU_QEMU_HARDDISK_drive-scsi8
    Kind:  by-path
    Links:
      /dev/disk/by-path/pci-0000:00:05.0-scsi-0:0:0:8
  Filesystem:
  Partitioned:  No
  Path:         /dev/sdi
Status:
  Claim State:  Unclaimed
  State:        Active
Events:         <none>
```

:question: ranjithr005:mayadata_m: 
Can we ensure the serial number of these disks are different. Using kubctl describe bd <bd-name> -n openebs, you will get the serial number of the disks.

:bulb: Wilf1rst
blockdevice-10cbf5d6bd8330920fcb3bdda1aa6a55
```yaml
$ kubectl -n openebs describe bd blockdevice-10cbf5d6bd8330920fcb3bdda1aa6a55
Name:         blockdevice-10cbf5d6bd8330920fcb3bdda1aa6a55
Namespace:    openebs
Labels:       kubernetes.io/hostname=k8snodv03p
              ndm.io/blockdevice-type=blockdevice
              ndm.io/managed=true
Annotations:  <none>
API Version:  openebs.io/v1alpha1
Kind:         BlockDevice
Metadata:
  Creation Timestamp:  2019-08-10T10:25:33Z
  Generation:          1
  Resource Version:    2257365
  Self Link:           /apis/openebs.io/v1alpha1/namespaces/openebs/blockdevices/blockdevice-10cbf5d6bd8330920fcb3bdda1aa6a55
  UID:                 2f820cd9-bb59-11e9-8221-0200005283a7
Spec:
  Capacity:
    Logical Sector Size:   512
    Physical Sector Size:  0
    Storage:               34359738368
  Details:
    Compliance:         
    Device Type:        
    Firmware Revision:  
    Model:              QEMU_HARDDISK
    Serial:             drive-scsi2
    Vendor:             QEMU
  Devlinks:
    Kind:  by-id
    Links:
      /dev/disk/by-id/scsi-0QEMU_QEMU_HARDDISK_drive-scsi2
    Kind:  by-path
    Links:
      /dev/disk/by-path/pci-0000:00:05.0-scsi-0:0:0:2
  Filesystem:
  Partitioned:  No
  Path:         /dev/sdc
Status:
  Claim State:  Unclaimed
  State:        Active
Events:         <none>
```

:spiral_notepad: blockdevice-11ebb418e80395b0378eca084e2a17e4
```yaml
$ kubectl -n openebs describe bd blockdevice-11ebb418e80395b0378eca084e2a17e4
Name:         blockdevice-11ebb418e80395b0378eca084e2a17e4
Namespace:    openebs
Labels:       kubernetes.io/hostname=k8snodv03p
              ndm.io/blockdevice-type=blockdevice
              ndm.io/managed=true
Annotations:  <none>
API Version:  openebs.io/v1alpha1
Kind:         BlockDevice
Metadata:
  Creation Timestamp:  2019-08-10T10:26:00Z
  Generation:          1
  Resource Version:    2257466
  Self Link:           /apis/openebs.io/v1alpha1/namespaces/openebs/blockdevices/blockdevice-11ebb418e80395b0378eca084e2a17e4
  UID:                 3f6d5cbb-bb59-11e9-8221-0200005283a7
Spec:
  Capacity:
    Logical Sector Size:   512
    Physical Sector Size:  0
    Storage:               34359738368
  Details:
    Compliance:         
    Device Type:        
    Firmware Revision:  
    Model:              QEMU_HARDDISK
    Serial:             drive-scsi3
    Vendor:             QEMU
  Devlinks:
    Kind:  by-id
    Links:
      /dev/disk/by-id/scsi-0QEMU_QEMU_HARDDISK_drive-scsi3
    Kind:  by-path
    Links:
      /dev/disk/by-path/pci-0000:00:05.0-scsi-0:0:0:3
  Filesystem:
  Partitioned:  No
  Path:         /dev/sdd
Status:
  Claim State:  Unclaimed
  State:        Active
Events:         <none>
```

:spiral_notepad: blockdevice-169068a2a396a22759d5bfa04f4a1b04
```yaml
$ kubectl -n openebs describe bd blockdevice-169068a2a396a22759d5bfa04f4a1b04
Name:         blockdevice-169068a2a396a22759d5bfa04f4a1b04
Namespace:    openebs
Labels:       kubernetes.io/hostname=k8snodv03p
              ndm.io/blockdevice-type=blockdevice
              ndm.io/managed=true
Annotations:  <none>
API Version:  openebs.io/v1alpha1
Kind:         BlockDevice
Metadata:
  Creation Timestamp:  2019-08-10T10:42:16Z
  Generation:          1
  Resource Version:    2260911
  Self Link:           /apis/openebs.io/v1alpha1/namespaces/openebs/blockdevices/blockdevice-169068a2a396a22759d5bfa04f4a1b04
  UID:                 854381b1-bb5b-11e9-8221-0200005283a7
Spec:
  Capacity:
    Logical Sector Size:   512
    Physical Sector Size:  0
    Storage:               34359738368
  Details:
    Compliance:         
    Device Type:        
    Firmware Revision:  
    Model:              QEMU_HARDDISK
    Serial:             drive-scsi7
    Vendor:             QEMU
  Devlinks:
    Kind:  by-id
    Links:
      /dev/disk/by-id/scsi-0QEMU_QEMU_HARDDISK_drive-scsi7
    Kind:  by-path
    Links:
      /dev/disk/by-path/pci-0000:00:05.0-scsi-0:0:0:7
  Filesystem:
  Partitioned:  No
  Path:         /dev/sdh
Status:
  Claim State:  Unclaimed
  State:        Active
Events:         <none>
```

:spiral_notepad: blockdevice-51e5c068c21f5186044957960dfbaefb
```yaml
$ kubectl -n openebs describe bd blockdevice-51e5c068c21f5186044957960dfbaefb
Name:         blockdevice-51e5c068c21f5186044957960dfbaefb
Namespace:    openebs
Labels:       kubernetes.io/hostname=k8snodv03p
              ndm.io/blockdevice-type=blockdevice
              ndm.io/managed=true
Annotations:  <none>
API Version:  openebs.io/v1alpha1
Kind:         BlockDevice
Metadata:
  Creation Timestamp:  2019-08-10T10:24:59Z
  Generation:          1
  Resource Version:    2257294
  Self Link:           /apis/openebs.io/v1alpha1/namespaces/openebs/blockdevices/blockdevice-51e5c068c21f5186044957960dfbaefb
  UID:                 1b720aa1-bb59-11e9-8221-0200005283a7
Spec:
  Capacity:
    Logical Sector Size:   512
    Physical Sector Size:  0
    Storage:               34359738368
  Details:
    Compliance:         
    Device Type:        
    Firmware Revision:  
    Model:              QEMU_HARDDISK
    Serial:             drive-scsi1
    Vendor:             QEMU
  Devlinks:
    Kind:  by-id
    Links:
      /dev/disk/by-id/scsi-0QEMU_QEMU_HARDDISK_drive-scsi1
    Kind:  by-path
    Links:
      /dev/disk/by-path/pci-0000:00:05.0-scsi-0:0:0:1
  Filesystem:
  Partitioned:  No
  Path:         /dev/sdb
Status:
  Claim State:  Unclaimed
  State:        Active
Events:         <none>
```

:spiral_notepad: blockdevice-9d0e2efbf255eec778874a9fd42f1a72
```yaml
$ kubectl -n openebs describe bd blockdevice-9d0e2efbf255eec778874a9fd42f1a72
Name:         blockdevice-9d0e2efbf255eec778874a9fd42f1a72
Namespace:    openebs
Labels:       kubernetes.io/hostname=k8snodv01p
              ndm.io/blockdevice-type=blockdevice
              ndm.io/managed=true
Annotations:  <none>
API Version:  openebs.io/v1alpha1
Kind:         BlockDevice
Metadata:
  Creation Timestamp:  2019-08-10T10:29:18Z
  Generation:          1
  Resource Version:    2258344
  Self Link:           /apis/openebs.io/v1alpha1/namespaces/openebs/blockdevices/blockdevice-9d0e2efbf255eec778874a9fd42f1a72
  UID:                 b5815988-bb59-11e9-8221-0200005283a7
Spec:
  Capacity:
    Logical Sector Size:   512
    Physical Sector Size:  0
    Storage:               34359738368
  Details:
    Compliance:         
    Device Type:        
    Firmware Revision:  
    Model:              QEMU_HARDDISK
    Serial:             drive-scsi4
    Vendor:             QEMU
  Devlinks:
    Kind:  by-id
    Links:
      /dev/disk/by-id/scsi-0QEMU_QEMU_HARDDISK_drive-scsi4
    Kind:  by-path
    Links:
      /dev/disk/by-path/pci-0000:00:05.0-scsi-0:0:0:4
  Filesystem:
  Partitioned:  No
  Path:         /dev/sde
Status:
  Claim State:  Unclaimed
  State:        Active
Events:         <none>
```

:spiral_notepad: blockdevice-c2b57cce00020b1d1562b9c469a949fa
```yaml
$ kubectl -n openebs describe bd blockdevice-c2b57cce00020b1d1562b9c469a949fa
Name:         blockdevice-c2b57cce00020b1d1562b9c469a949fa
Namespace:    openebs
Labels:       kubernetes.io/hostname=k8snodv01p
              ndm.io/blockdevice-type=blockdevice
              ndm.io/managed=true
Annotations:  <none>
API Version:  openebs.io/v1alpha1
Kind:         BlockDevice
Metadata:
  Creation Timestamp:  2019-08-10T10:29:40Z
  Generation:          1
  Resource Version:    2258429
  Self Link:           /apis/openebs.io/v1alpha1/namespaces/openebs/blockdevices/blockdevice-c2b57cce00020b1d1562b9c469a949fa
  UID:                 c2941896-bb59-11e9-8221-0200005283a7
Spec:
  Capacity:
    Logical Sector Size:   512
    Physical Sector Size:  0
    Storage:               34359738368
  Details:
    Compliance:         
    Device Type:        
    Firmware Revision:  
    Model:              QEMU_HARDDISK
    Serial:             drive-scsi5
    Vendor:             QEMU
  Devlinks:
    Kind:  by-id
    Links:
      /dev/disk/by-id/scsi-0QEMU_QEMU_HARDDISK_drive-scsi5
    Kind:  by-path
    Links:
      /dev/disk/by-path/pci-0000:00:05.0-scsi-0:0:0:5
  Filesystem:
  Partitioned:  No
  Path:         /dev/sdf
Status:
  Claim State:  Unclaimed
  State:        Active
Events:         <none>
```

:spiral_notepad: blockdevice-dae3cec149cfd5fc5c6826caa0650e21
```yaml
$ kubectl -n openebs describe bd blockdevice-dae3cec149cfd5fc5c6826caa0650e21
Name:         blockdevice-dae3cec149cfd5fc5c6826caa0650e21
Namespace:    openebs
Labels:       kubernetes.io/hostname=k8snodv01p
              ndm.io/blockdevice-type=blockdevice
              ndm.io/managed=true
Annotations:  <none>
API Version:  openebs.io/v1alpha1
Kind:         BlockDevice
Metadata:
  Creation Timestamp:  2019-08-10T10:30:04Z
  Generation:          1
  Resource Version:    2258639
  Self Link:           /apis/openebs.io/v1alpha1/namespaces/openebs/blockdevices/blockdevice-dae3cec149cfd5fc5c6826caa0650e21
  UID:                 d14cde9e-bb59-11e9-8221-0200005283a7
Spec:
  Capacity:
    Logical Sector Size:   512
    Physical Sector Size:  0
    Storage:               34359738368
  Details:
    Compliance:         
    Device Type:        
    Firmware Revision:  
    Model:              QEMU_HARDDISK
    Serial:             drive-scsi6
    Vendor:             QEMU
  Devlinks:
    Kind:  by-id
    Links:
      /dev/disk/by-id/scsi-0QEMU_QEMU_HARDDISK_drive-scsi6
    Kind:  by-path
    Links:
      /dev/disk/by-path/pci-0000:00:05.0-scsi-0:0:0:6
  Filesystem:
  Partitioned:  No
  Path:         /dev/sdg
Status:
  Claim State:  Unclaimed
  State:        Active
Events:         <none>
```

:spiral_notepad: blockdevice-db1254ebd777a99e6b9b5626358c7038
```yaml
$ kubectl -n openebs describe bd blockdevice-db1254ebd777a99e6b9b5626358c7038
Name:         blockdevice-db1254ebd777a99e6b9b5626358c7038
Namespace:    openebs
Labels:       kubernetes.io/hostname=k8snodv01p
              ndm.io/blockdevice-type=blockdevice
              ndm.io/managed=true
Annotations:  <none>
API Version:  openebs.io/v1alpha1
Kind:         BlockDevice
Metadata:
  Creation Timestamp:  2019-08-10T10:43:45Z
  Generation:          1
  Resource Version:    2261226
  Self Link:           /apis/openebs.io/v1alpha1/namespaces/openebs/blockdevices/blockdevice-db1254ebd777a99e6b9b5626358c7038
  UID:                 ba9b82df-bb5b-11e9-8221-0200005283a7
Spec:
  Capacity:
    Logical Sector Size:   512
    Physical Sector Size:  0
    Storage:               34359738368
  Details:
    Compliance:         
    Device Type:        
    Firmware Revision:  
    Model:              QEMU_HARDDISK
    Serial:             drive-scsi8
    Vendor:             QEMU
  Devlinks:
    Kind:  by-id
    Links:
      /dev/disk/by-id/scsi-0QEMU_QEMU_HARDDISK_drive-scsi8
    Kind:  by-path
    Links:
      /dev/disk/by-path/pci-0000:00:05.0-scsi-0:0:0:8
  Filesystem:
  Partitioned:  No
  Path:         /dev/sdi
Status:
  Claim State:  Unclaimed
  State:        Active
Events:         <none>
```

Wilf1rst:
:spiral_notepad: Here's all my bd description :slightly_smiling_face:

:nerd_face: vitta-at-mayadata:mayadata_m: 
NDM detects that disks are different by looking at Model, Serial and vendor..
in this case, if I'm not wrong, one of the disk in each node is having serial as 'drive-scsi*'

:bulb: vitta-at-mayadata:mayadata_m: 
can we make sure that disks in entire cluster (across nodes) are having different serial numbers?

Wilf1rst
:thinking: Indeed I only see one drive-scsi serial between 1 and 8:
```yaml
$ kubectl -n openebs describe bd |grep -A30 "Name:         blockdevice-"|grep -i serial|sort
    Serial:             drive-scsi1
    Serial:             drive-scsi2
    Serial:             drive-scsi3
    Serial:             drive-scsi4
    Serial:             drive-scsi5
    Serial:             drive-scsi6
    Serial:             drive-scsi7
    Serial:             drive-scsi8
```

:spiral_notepad: As my virtualisation host is a **Proxmox server**, I assume on each VMs there's a mounted disk with serial drive-scsi1, drive-scsi2 and so on ...So as it seams that the serial is binded on the scsi bus / device I've to set another scsi device if I understand well:
Like
```
* 1 to 8 for my node1
* 9 to 16 for my node2
* 17 to 24 form my node3 (edited)
```

Wilf1rst 
:bulb: Thanks i better understand how openebs detect bd

:nerd_face: vitta-at-mayadata:mayadata_m: 
perfect.. if there is another possibility of enabling WWN on virtual machines (similar to vsphere), it would be better

:question: vitta-at-mayadata:mayadata_m:  
BTW, whats the virtualization platform you are using?

Wilf1rst 
:spiral_notepad: Here i use [proxmox](https://www.proxmox.com/en/).
```
proxmox.comproxmox.com
Proxmox - powerful open-source server solutions
Proxmox offers the server virtualization management platform Proxmox VE, and the Proxmox Mail Gateway an antispam and antivirus solution for mail server protection.
```

:spiral_notepad: vitta-at-mayadata:mayadata_m:
thanks @Wilf1rst.. will go thru that to see options for enabling WWNs.. cc: @akhilerm 

:spiral_notepad: Wilf1rst
I use to works with vSphere at work but hereI use proxmox for it's LXC support and i don't see a simple option to set my WWN in my disk creation. I will check on proxmox doc

vitta-at-mayadata:mayadata_m: 
:+1: ok.. if we make sure different serials for all disks, it works

Wilf1rst
:+1: To add to the information I just found that I can add VirtIO Block disk with the same device number in the proxmox console and the two are seen correctly by openebs.

Wilf1rst
By the way there's no serial set:
```yaml
$ kubectl -n openebs describe bd blockdevice-e3860833c7243177f4a60fe0b9a01aac
Name:         blockdevice-e3860833c7243177f4a60fe0b9a01aac
Namespace:    openebs
Labels:       kubernetes.io/hostname=k8snodv01p
              ndm.io/blockdevice-type=blockdevice
              ndm.io/managed=true
Annotations:  <none>
API Version:  openebs.io/v1alpha1
Kind:         BlockDevice
Metadata:
  Creation Timestamp:  2019-08-10T12:22:34Z
  Generation:          1
  Resource Version:    2284595
  Self Link:           /apis/openebs.io/v1alpha1/namespaces/openebs/blockdevices/blockdevice-e3860833c7243177f4a60fe0b9a01aac
  UID:                 882f3157-bb69-11e9-8221-0200005283a7
Spec:
  Capacity:
    Logical Sector Size:   512
    Physical Sector Size:  0
    Storage:               34359738368
  Details:
    Compliance:         
    Device Type:        
    Firmware Revision:  
    Model:              
    Serial:             
    Vendor:             
  Devlinks:
    Kind:  by-path
    Links:
      /dev/disk/by-path/pci-0000:00:0b.0
      /dev/disk/by-path/virtio-pci-0000:00:0b.0
  Filesystem:
  Partitioned:  No
  Path:         /dev/vda
Status:
  Claim State:  Unclaimed
  State:        Active
Events:         <none>
```

:spiral_notepad: vitta-at-mayadata:mayadata_m:
checking with team on the steps of identifying the disks by NDM

Wilf1rst 
:heart: Thanks a lot. Your product rocks and you the dev are very helpful. Thank you again

:spiral_notepad: vitta-at-mayadata:mayadata_m: 
this might help - https://docs.openebs.io/docs/next/ndm.html#ndm-daemonset-functions
however, I see that this can be improved..

**Comment by AmitKumarDas** [comment]
cc @vishnuitta @ranjithwingrider @akhilerm @wilf1rst

**Comment by vantonio** [comment]
I was able to solve this by modifying the configuration file of a vm:

edit conf file:
````vi /etc/pve/qemu-server/101.conf````

add a serial number:
```scsi1: images:vm-101-disk-1,cache=writeback,discard=on,size=120G,ssd=1```**,serial=5fb20ba17c2f**

restart vm:
```qm shutdown 101 && qm start 101```

we now have a unique disk id:
```
$ ls -lah /dev/disk/by-id
lrwxrwxrwx 1 root root   9 Aug 30 00:20  scsi-0QEMU_QEMU_HARDDISK_5fb20ba17c2f -> ../../sdb
```

Hope it helps!

**Comment by wilf1rst** [comment]
Great !!

On my side I resolved this issue by using `VirtIO block` disks attached to my VM on Proxmox Virtual Machine Hardware configuration.

**Comment by footplus** [comment]
This seems to be related to a mis-detection in NDM of some kinds of Virtual disks, and there is already an issue there: https://github.com/openebs/node-disk-manager/issues/188 . I think that adding a serial is one option, but at cloud providers the user is not in control, and the issue may be better fixed in the same way.

**Comment by akhilerm** [comment]
Yes @footplus . We are in the design phase of creating a generic solution for disk identification to work on all platforms. It would be great if you can share your inputs on #2666 this design PR.

**Comment by akhilerm** [comment]
This issue has been fixed with openebs 2.0, where a new UUID algorithm for blockdevices is used. Ref : [Change-summary](https://github.com/openebs/openebs/wiki/Change-Summary-v2.0#new-capabilities)
